client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +49,41 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    @Getter
+    private final boolean isSubscriber;
+    @Getter
+    private final boolean autoTruncateAtLastCheckpoint;
+
    public static class ReaderGroupConfigBuilder implements ObjectBuilder<ReaderGroupConfig> {
        private long groupRefreshTimeMillis = 3000; //default value
        private long automaticCheckpointIntervalMillis = 30000; //default value
        // maximum outstanding checkpoint request that is allowed at any given time.
        private int maxOutstandingCheckpointRequest = 3; //default value
+       private boolean subscribedForRetention = false; //default value
+       private boolean autoPublishLastCheckpoint = false; //default value
+
+       /**
+        * Makes the reader group a subscriber reader group. During CBR (Consumption Based Retention),
+        * the subscriber reader group's positions are considered during truncation. The stream will truncate
+        * to the last checkpoint position of this group.
+        *
+        * @return Reader group config builder.
+        */
+       public ReaderGroupConfigBuilder isSubscribedForRetention() {
+           this.subscribedForRetention = true;
+           return this;
+       }
+
+       /**
+        * Allows for every checkpoint initiated by this reader group (automatic or manual)
+        * to be published as a truncation point to the controller, if the reader group is a subscriber reader group.
+        *
+        * @return Reader group config builder.
+        */
+       public ReaderGroupConfigBuilder autoPublishCheckpoint() {
```

Comment:
method should be called `autoPublishLastCheckpoint` too.

Reply:
Changed method name